src/libstd/net/tcp.rs

```diff
@@ -154,8 +164,7 @@ impl TcpStream {
     /// # Note
     ///
     /// Some platforms do not provide access to the current timeout.
-    #[unstable(feature = "socket_timeout", reason = "RFC 1047 - recently added",
-               issue = "27773")]
+    #[stable(feature = "socket_timeout", since = "1.4.0")]
```

Comment:
This works on Windows as far as I can tell: https://github.com/rust-lang/rust/blob/master/src/libstd/net/tcp.rs#L909